tools/cosmovisor/cmd/cosmovisor/run.go

```diff
@@ -18,7 +27,7 @@ var runCmd = &cobra.Command{
 
 // run runs the configured program with the given args and monitors it for upgrades.
 func run(args []string, options ...RunOption) error {
-	cfg, err := cosmovisor.GetConfigFromEnv()
+	cfg, err := cosmovisor.GetConfigFromFile(os.Getenv(cfgENV))
```

Comment:
I don't get this, why cannot we use the `--config` flag from cosmovisor.
I don't like that you need another env variable just for the config name.

Reply:
As you can see [here](https://github.com/cosmos/cosmos-sdk/pull/19995/files/b2f5594aec6976f04d13f650de3121f5c4390146#diff-ce38c75be4a2d1e7babbf8d1a0e53b5b6e5753e4b2d9427460b00844b976d378R22) run command has disabled flag parsing.
So I didn't want to change the existing behaviour of the run command.

Let me know your thoughts on this I will update it.

Reply:
That is true. Then I think we should maybe rename the --config flag so we are sure it won't clash with an app flag, maybe --cosmovisor-config. Then I'd personally say we should parse that one for consistency here too.
It will a tiny be less nice to implement but the UX will be better.
Are you willing to do that? Otherwise happy to do that myself in a follow up.

Reply:
Okay.
So you mean we should replace the `config` flag with `cosmovisor-config` in root commands and enable flag parsing in run command.

Reply:
Yes, but you shouldn't enable flag parsing as you pass the app command there. You'll need to manually parse --cosmovisor-config there.

Reply:
Done.